### PR TITLE
Fix replacements order

### LINKED ISSUE #12134: When replacing questions, the replacements should be placed in the positions of the questions which were selected to be replaced.
## Overview

During question replacement in quiz creation, the replacements are being slotted to the bottom of the list rather than put into the place of the questions which were selected to be replaced.

This [`handleReplacement`](https://github.com/nucleogenesis/kolibri/blob/0b2aa57e7c69904a6e43e0e5aa3823d780444541/kolibri/plugins/coach/assets/src/composables/useQuizCreation.js#L169-L177) function should be refactored so that the position of the questions to be removed are replaced with the user's selected replacements.

This function currently just gathers a list of the questions which are _not_ being replaced. Then it just pops the new questions on the end. I think that updating the `filter` to a `map` where we either return the first replacement or the "activeQuestion" we're getting in the iteration. **I _think_ an approach like this can avoid getting complicated but don't feel committed to this if it doesn't pan out.**

_Note how the questions I selected for replacement are at the bottom rather than where I'd selected the questions I wanted replaced_

https://github.com/learningequality/kolibri/assets/6356129/4bfaff6b-6edb-4991-b8c4-3718ab548e60

### PR DESCRIPTION
**Pull request by AlexVelezLl**
## Summary

This PR maps the question replacements to the order in which the selected questions are.

## References

Closes #12134.

## Reviewer guidance

See $12134

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/coach/assets/src/composables/useQuizCreation.js [hunk 1]
```diff
@@ -167,14 +167,16 @@ export default function useQuizCreation() {
...
   function handleReplacement() {
-    const questionsNotSelectedToBeReplaced = activeQuestions.value.filter(
-      question => !selectedActiveQuestions.value.includes(question.id)
-    );
+    const questions = activeQuestions.value.map(question => {
+      if (selectedActiveQuestions.value.includes(question.id)) {
+        return replacements.value.shift();
+      }
+      return question;
+    });
```
Comment (nucleogenesis) [inline kolibri/plugins/coach/assets/src/composables/useQuizCreation.js:175]: Love this mapping and shifting of the replacements list.

**Review by nucleogenesis — APPROVED** [review]
Tested this locally and code LGTM!

**Comment by pcenov** [comment]
No issues observed while manually testing. Good to go! Thanks @AlexVelezLl

**Review by radinamatic — APPROVED** [review]
Manual QA passes, good to go! 💯 👏🏽 :shipit: 
Thank you @AlexVelezLl !

**Pull request merged**